### ISSUE TITLE
Issues with generic type alias that uses a TypeVar bound to a type

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3401]
### Discussed in https://github.com/microsoft/pyright/discussions/3390

<div type='discussions-op-text'>

<sup>Originally posted by **povilasb** April 27, 2022</sup>
I'm having hard time to grasp if this is an issue with pyright or I'm misinterpreting Python type system.
Consider this small example:

```py
from typing import TypeVar, Sequence, Type

_T = TypeVar("_T", bound=Type[Exception])
_MaybeSequence = _T | Sequence[_T]


class HttpError(Exception):
    pass


def retry(errs: _MaybeSequence[type[Exception]]):
    pass


# both of these cases fail the type checks
retry(HttpError)
retry(Exception)
```

pyright (`1.1.240`) errors with

```sh
  /tmp/main.py:15:7 - error: Argument of type "Type[HttpError]" cannot be assigned to parameter "errs" of type "_MaybeSequence[Type[Exception]]" in function "retry"
    Type "Type[HttpError]" cannot be assigned to type "_MaybeSequence[Type[Exception]]"
      "Type[type]" is incompatible with "Type[Exception]"
      "Type[type]" is incompatible with "Type[Sequence[Type[Exception]]]" (reportGeneralTypeIssues)
```

"Type[type]" sounds suspicious.

Maybe subtypes of types are not even supposed to work this way?

I can check them during the runtime though
```py
print(issubclass(HttpError, Exception))
True
```
Also, `pyright` in this case says
```py
error: Unnecessary issubclass call; "Type[HttpError]" is always a subclass of "Exception" (reportUnnecessaryIsInstance)
```</div>

**Comment by erictraut** [comment]
I've figured out the root cause of this problem. It will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.242, which I just published. It will also be included in the next release of pylance.
